py/bitbox02/bitbox02/bitbox02.py

```diff
@@ -229,19 +229,22 @@ def __init__(
         send_msg = noise.write_message()
         assert noise.handshake_finished
         pairing_code = base64.b32encode(noise.get_handshake_hash()).decode("ascii")
-        show_pairing_callback(
-            "{} {}\n{} {}".format(
-                pairing_code[:5], pairing_code[5:10], pairing_code[10:15], pairing_code[15:20]
-            )
-        )
         response = self._query(send_msg)
 
         # Can be set to False if the remote static pubkey was previously confirmed.
         pairing_verification_required_by_host = True
 
         pairing_verification_required_by_device = response == b"\x01"
         if pairing_verification_required_by_host or pairing_verification_required_by_device:
-            pairing_response = self._query(OP_I_CAN_HAS_PAIRIN_VERIFICASHUN)
+            cid = u2fhid.generate_cid()
+            u2fhid.write(self.device, OP_I_CAN_HAS_PAIRIN_VERIFICASHUN, HWW_CMD, cid)
+            show_pairing_callback(
```

Comment:
the callback should be extended also to get a user response (accept, reject) and handle that properly.

now or in another PR?

Reply:
I can do it now, but not sure how to do it properly. Where do you do that in the bitbox02-api-go package?

Reply:
the rejection should probably just raise a UserRejectedException.

Reply:
the callback should also not take a formatted string. It should take `pairing_code[:20]`. Formatting should be done in the application.

Reply:
I think I disagree. The formatting with with spaces and two lines is fixed by the bb02, any client has to match it. Re-doing the same code in every client does not add any value.

Reply:
I raise an exception now. Please merge the other PR first, I already added a `self.close()` here in anticipation.